client/dashboard/customizable.js

```diff
@@ -302,7 +302,10 @@ class CustomizableDashboard extends Component {
 	render() {
 		const { query, taskListHidden, taskListComplete } = this.props;
 
-		const isTaskListEnabled = isOnboardingEnabled() && ! taskListHidden;
+		const isTaskListEnabled =
+			isOnboardingEnabled() &&
+			! taskListHidden &&
+			! window.wcAdminFeatures.homepage;
```

Comment:
@jeffstieler if #4418 is merged first this should be updated to `! getFeatureFlag( 'homepage' );`